l10n_ch_credit_control_dunning_fees_qr_report/models/credit_control_communication_report.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+import io
+from collections import OrderedDict
+
+from odoo import api, models
+
+
+class ReportSwissQR(models.AbstractModel):
+    _inherit = "report.l10n_ch.qr_report_main"
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        """Replace QR URL to consider dunning fees"""
+        self = self.with_context(credit_control_run_print_report=True)
+        res = super()._get_report_values(docids, data=data)
+        dunning_fees_qr_amounts = self.env.context.get("qr_dunning_fees_amounts", {})
+        if not dunning_fees_qr_amounts:
+            return res
+        qr_code_urls = {}
+        invoices = self.env["account.move"].browse(docids)
+        for invoice in invoices:
+            if invoice.id not in dunning_fees_qr_amounts:
+                continue
```

Comment:
```suggestion
        invoices = self.env["account.move"].browse(list(dunning_fees_qr_amounts.keys()))
        for invoice in invoices:
```
You know already the ones to browse no?